cpp/src/strings/case.cu

```diff
@@ -345,22 +427,26 @@ std::unique_ptr<column> convert_case(strings_column_view const& input,
                                cudf::detail::copy_bitmask(input.parent(), stream, mr));
   }
 
-  // Check if the input contains any multi-byte characters.
+  // Check if the input contains special multi-byte characters where the case equivalent
+  // has a different number of bytes. There are only 100 such special characters.
   // This check incurs ~20% performance hit for smaller strings and so we only use it
   // after the threshold check above. The check makes very little impact for long strings
-  // but results in a large performance gain when the input contains only single-byte characters.
+  // but results in a large performance gain when the input contains no special characters.
+  constexpr int64_t bytes_per_thread = 4;
   cudf::detail::device_scalar<int64_t> mb_count(0, stream);
-  // cudf::detail::grid_1d is limited to size_type elements
-  auto const num_blocks = util::div_rounding_up_safe(chars_size / bytes_per_thread, block_size);
-  // we only need to check every other byte since either will contain high bit
-  has_multibytes_kernel<<<num_blocks, block_size, 0, stream.value()>>>(
-    input_chars, first_offset, last_offset, mb_count.data());
+  auto const grid = cudf::detail::grid_1d(chars_size, block_size, bytes_per_thread);
+  mismatch_multibytes_kernel<bytes_per_thread>
+    <<<grid.num_blocks, grid.num_threads_per_block, 0, stream.value()>>>(
+      input_chars, first_offset, last_offset, mb_count.data());
   if (mb_count.value(stream) == 0) {
-    // optimization for ASCII-only case: copy the input column and inplace replace each character
+    // optimization for the non-special case;
+    // copying the input column automatically handles normalizing sliced inputs
+    // and nulls properly but also does incur a wasteful chars copy too
```

Comment:
If I'm understanding this right and this comment is about an alternative approach that is not used:
```suggestion
    // copying the input column would automatically handle normalizing sliced inputs
    // and nulls properly but would also incur a wasteful chars copy
```

Reply:
I feel my original wording is more accurate. I feel like `would` here conveys it should be doesn't yet. 
Perhaps `will` is a better alternative but I'm still not quite sure what was the difficulty with the original text.

Reply:
My difficulty was with understanding that the comment describes an approach that's not used. There's nothing in the code about column copying, so the comment seems a bit disconnected. Making it explicit that the comment explains why we're not just copying would be helpful to make the code more readable IMO.

Reply:
This line here:
```
auto result  = std::make_unique<column>(input.parent(), stream, mr);
```
Makes a copy of the input column. It also has the nice side-effect of normalizing out any slicey-ness.
We use this copy to place the output result into.

Reply:
I misunderstood the comment and the code with it; I'll take the blame and leave the comment alone :)

Reply:
Now I'm considering rewording this comment to include my new word: _slicey-ness_